x/evidence/keeper/infraction.go

```diff
@@ -24,11 +24,20 @@ import (
 //
 // TODO: Some of the invalid constraints listed above may need to be reconsidered
 // in the case of a lunatic attack.
+
 func (k Keeper) handleEquivocationEvidence(ctx context.Context, evidence *types.Equivocation) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	logger := k.Logger(sdkCtx)
 	consAddr := evidence.GetConsensusAddress()
 
+	if val := k.stakingKeeper.ValidatorByConsAddr(sdkCtx, consAddr.Bytes()); val == nil {
```

Comment:
should this be handled in ValidatorByConsAddr? so that we dont have to have nil checks, its just if the key doesnt exist check the mapped conskey if nothing then it doesnt exist